lib/internal/errors.js

```diff
@@ -175,6 +192,7 @@ E('ERR_SOCKET_BAD_TYPE',
 E('ERR_SOCKET_CANNOT_SEND', 'Unable to send data');
 E('ERR_SOCKET_BAD_PORT', 'Port should be > 0 and < 65536');
 E('ERR_SOCKET_DGRAM_NOT_RUNNING', 'Not running');
+E('ERR_UNKNOWN_ENCODING', (encoding) => `Unknown encoding: ${encoding}`);
```

Comment:
I am not sure if the error codes should be more generic or not but I think it would make sense if they are. In this case the `ERR_INVALID_OPT_VALUE` type could also be used as such.